include/boost/program_options/parsers.hpp

```diff
@@ -120,7 +120,10 @@ namespace boost { namespace program_options {
         basic_command_line_parser(const std::vector<
                                   std::basic_string<charT> >& args);
         /** Creates a command line parser for the specified arguments
-            list. The parameters should be the same as passed to 'main'.
+            list. The parameters should be the same as passed to 'main', meaning:
+            @param argc Must be non-zero i.e. >= 0
```

Comment:
Two things. Non-zero is >0. Secondly, I think the standard mandates a argc>=1 and argv[0] is the program name on all known platforms

Reply:
I just had a discussion about that prior with a theoretical physicist: according to him, everyone in the math world defines non-zero with >=0. Also got references: https://mathworld.wolfram.com/Nonnegative.html 

Math world besides, as mentioned in bug report, cpp standard seem to share this definition too. As it defines the behavior for argc == 0 cases.

Reply:
To the second point, see comment below

Reply:
> Also got references: https://mathworld.wolfram.com/Nonnegative.html

That's Nonnegative, though

Reply:
Resolved?

Reply:
Sure (I cannot find a button to mark as resolved)

Reply:
Well for some reason I can resolve it. But this of course only makes sense, if you would have :-)